ServerCore/Pages/Teams/AutoTeam.razor.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components;
+using Microsoft.EntityFrameworkCore;
+using ServerCore.DataModel;
+using ServerCore.Helpers;
+using ServerCore.ModelBases;
+
+namespace ServerCore.Pages.Teams
+{
+    public partial class AutoTeam
+    {
+        [Parameter]
+        public Event Event { get; set; }
+
+        [Parameter]
+        public int LoggedInUserId { get; set; }
+
+        [Parameter]
+        public EventRole EventRole { get; set; }
+
+        [Inject]
+        public PuzzleServerContext _context { get; set; }
+
+        [Inject]
+        public NavigationManager NavigationManager { get; set; }
+
+        [Required]
+        public AutoTeamType? PlayerType { get; set; } = null;
+
+        bool CantCreateTeam { get; set; }
+
+        public async Task OnSubmit()
+        {
+            CantCreateTeam = false;
+
+            int existingAutoTeam = await (from team in _context.Teams
+                                           join tempTeamMember in _context.TeamMembers on team.ID equals tempTeamMember.Team.ID into teamMembers
+                                           from teamMember in teamMembers.DefaultIfEmpty()
+                                           where team.Event == Event && team.AutoTeamType == PlayerType
+                                           group teamMember by team.ID into teamGroup
+                                           where teamGroup.Count() < Event.MaxTeamSize
```

Comment:
We should leave some extra space here, so that autoplayers can add +1s to their team, and so the last autoteam of like 2 people has somewhere they can be merged. This could either be some hardcoded number like 2, or an event property.

Reply:
We also discussed using MaxTeamSize * 0.2, rounded to the nearest integer, which is 2 for 12 person teams and 1 for 4 person teams.